tests/nodescaling/scale_test.go

```diff
@@ -0,0 +1,165 @@
+	cleanup "github.com/rancher/tfp-automation/framework/cleanup"
+	"github.com/rancher/tfp-automation/tests/extensions/provisioning"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ScaleTestSuite struct {
+	suite.Suite
+	client             *rancher.Client
+	standardUserClient *rancher.Client
+	session            *session.Session
+	terraformConfig    *config.TerraformConfig
+	clusterConfig      *config.TerratestConfig
+	terraformOptions   *terraform.Options
+}
+
+func (r *ScaleTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	r.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(r.T(), err)
+
+	r.client = client
+
+	enabled := true
+	var testuser = namegen.AppendRandomString("testuser-")
+	var testpassword = password.GenerateUserPassword("testpass-")
+	user := &management.User{
+		Username: testuser,
+		Password: testpassword,
+		Name:     testuser,
+		Enabled:  &enabled,
+	}
+
+	newUser, err := users.CreateUserWithRole(client, user, "user")
+	require.NoError(r.T(), err)
+
+	newUser.Password = user.Password
+
+	standardUserClient, err := client.AsUser(newUser)
+	require.NoError(r.T(), err)
+
+	r.standardUserClient = standardUserClient
+
+	terraformConfig := new(config.TerraformConfig)
+	ranchFrame.LoadConfig("terraform", terraformConfig)
+
+	r.terraformConfig = terraformConfig
+
+	clusterConfig := new(config.TerratestConfig)
+	ranchFrame.LoadConfig("terratest", clusterConfig)
+
+	r.clusterConfig = clusterConfig
+
+	terraformOptions, err := framework.Setup(r.T())
+	require.NoError(r.T(), err)
+
+	r.terraformOptions = terraformOptions
+}
+
+func (r *ScaleTestSuite) TestScale() {
+	nodeRolesAll := []config.Nodepool{config.AllRolesNodePool}
+	scaleUpRolesAll := []config.Nodepool{config.ScaleUpAllRolesNodePool}
+	scaleDownRolesAll := []config.Nodepool{config.ScaleDownAllRolesNodePool}
+
+	nodeRolesShared := []config.Nodepool{config.EtcdControlPlaneNodePool, config.WorkerNodePool}
+	scaleUpRolesShared := []config.Nodepool{config.ScaleUpEtcdControlPlaneNodePool, config.ScaleUpWorkerNodePool}
+	scaleDownRolesShared := []config.Nodepool{config.ScaleUpEtcdControlPlaneNodePool, config.WorkerNodePool}
+
+	nodeRolesDedicated := []config.Nodepool{config.EtcdNodePool, config.ControlPlaneNodePool, config.WorkerNodePool}
+	scaleUpRolesDedicated := []config.Nodepool{config.ScaleUpEtcdNodePool, config.ScaleUpControlPlaneNodePool, config.ScaleUpWorkerNodePool}
+	scaleDownRolesDedicated := []config.Nodepool{config.ScaleUpEtcdNodePool, config.ScaleUpControlPlaneNodePool, config.WorkerNodePool}
+
+	tests := []struct {
+		name               string
+		nodeRoles          []config.Nodepool
+		scaleUpNodeRoles   []config.Nodepool
+		scaleDownNodeRoles []config.Nodepool
+		client             *rancher.Client
+	}{
+		{"Scaling 1 Node all roles " + config.StandardClientName.String(), nodeRolesAll, scaleUpRolesAll, scaleDownRolesAll, r.standardUserClient},
```

Comment:
nit; I still feel like these could have better names. I know we keep the names fairly short, but a little more info would be nice. i.e `Scale 1 node all roles -> 5 nodes -> 3 nodes` or something like that